### ISSUE TITLE
Weird/incorrect behaviour on MacOS 

### ISSUE THREAD

**Issue by ghost** [RedisLabs/memtier_benchmark#131]
When running `memtier_benchmark` on MacOS, after the benchmark completes I always see only `2` keys in Redis. I tried using different parameters, but no matter what I always end up with `2` keys in the database. One of the keys is always `memtier-0`. 

```
> ./memtier_benchmark --version
memtier_benchmark 1.3.0
Copyright (C) 2011-2020 Redis Labs Ltd.
....

> ./memtier_benchmark

4         Threads
50        Connections per thread
10000     Requests per client


ALL STATS
=========================================================================
Type         Ops/sec     Hits/sec   Misses/sec      Latency       KB/sec
-------------------------------------------------------------------------
Sets         9399.85          ---          ---      1.90700       693.97
Gets        93895.21     93836.28        58.93      1.93700      6458.41
Waits           0.00          ---          ---      0.00000          ---
Totals     103295.06     93836.28        58.93      1.93500      7152.39
```

`redis-cli` output after the benchmark:

```
127.0.0.1:6379> DBSIZE
(integer) 2
127.0.0.1:6379> keys *
1) "memtier-0"
2) "memtier-7664357"
127.0.0.1:6379>
```   

When I use a Linux machine, then it works fine:

```
127.0.0.1:6379> DBSIZE
(integer) 910
127.0.0.1:6379> keys *
  1) "memtier-1096872"
  2) "memtier-1283809"
  3) "memtier-3008385"
  4) "memtier-4618771"
  5) "memtier-643398"
  6) "memtier-27183"
  7) "memtier-512393"
  8) "memtier-6877537"
  9) "memtier-44808"
 10) "memtier-6644065"
 11) "memtier-2825641"
 ...
```

Is `memtier_benchmark` supposed to work on MacOS? Or am I missing something?

**Comment by yaacovhazan-Redislabs** [comment]
Hi, @mattiabi thank you for raising the issue, can you please run it again with the --key-pattern=S:S option? did you get the same wired results (only 2 keys)?

**Comment by mattiabi** [comment]
Hi @yaacovhazan-Redislabs; 

With `--key-pattern=S:S` I got 910 keys, similar to result on Linux. Does the default behaivour change while running on Linux and MacOS?

**Comment by yaacovhazan-Redislabs** [comment]
No, the default for all OS's is --key-pattern=R:R which mean Random ... so maybe something wrong with the way we produced random keys on MacOS

**Comment by jdeppe-pivotal** [comment]
I can confirm that I'm seeing this as well. Here is a short run with debug output:
```
$> memtier_benchmark -s 10.32.X.Y --ratio 1:1  --hide-histogram -t 1 -c 1 -n 10 --debug
Writing results to stdout
[RUN #1] Preparing benchmark client...
client.cpp:111: new client 0x7fa80fd043c0 successfully set up.
[RUN #1] Launching threads now...
shard_connection.cpp:555: server 10.32.108.85:6379: SET key=[memtier-0] value_len=32 expiry=0
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): +OK, 0 hits, 1 misses
shard_connection.cpp:568: server 10.32.108.85:6379: GET key=[memtier-7664357]
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): $32, 1 hits, 0 misses
shard_connection.cpp:555: server 10.32.108.85:6379: SET key=[memtier-7664357] value_len=32 expiry=0
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): +OK, 0 hits, 1 misses
shard_connection.cpp:568: server 10.32.108.85:6379: GET key=[memtier-0]
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): $32, 1 hits, 0 misses
shard_connection.cpp:555: server 10.32.108.85:6379: SET key=[memtier-0] value_len=32 expiry=0
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): +OK, 0 hits, 1 misses
shard_connection.cpp:568: server 10.32.108.85:6379: GET key=[memtier-7664357]
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): $32, 1 hits, 0 misses
shard_connection.cpp:555: server 10.32.108.85:6379: SET key=[memtier-0] value_len=32 expiry=0
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): +OK, 0 hits, 1 misses
shard_connection.cpp:568: server 10.32.108.85:6379: GET key=[memtier-7664357]
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): $32, 1 hits, 0 misses
shard_connection.cpp:555: server 10.32.108.85:6379: SET key=[memtier-7664357] value_len=32 expiry=0
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): +OK, 0 hits, 1 misses
shard_connection.cpp:568: server 10.32.108.85:6379: GET key=[memtier-0]
shard_connection.cpp:423: server 10.32.108.85:6379: handled response (first line): $32, 1 hits, 0 misses
client.cpp:219: nothing else to do, test is finished.
```

**Comment by yossigo** [comment]
@YaacovHazan Is this a duplicate of #158?

**Comment by YaacovHazan** [comment]
Closed, duplicate of #158
